### PR TITLE
Import: fix media duplication issue

### PR DESCRIPTION
**Pull request by ouikhuan**
Fixes Automattic/dotcom-forge#2016

## Proposed changes:

* When we upload images now they'll be created even if it's already existed. This will cause unnecessary duplication in the media library. This PR adds a check before we create the media to prevent duplication.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

See issue above
## Does this pull request change what data or activity we track or use?

No
## Testing instructions:

* You can spin up a JN site using [this branch](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack=fix/media-duplication&wp-debug-log) or spin up your local WP and your Jurrasic tube, you can do it by using `jetpack docker jt-up`. (If you're not familiar with this, you can check [the doc here](https://github.com/Automattic/jetpack/blob/trunk/tools/docker/README.md#jurassic-tube-tunneling-service).
* You can follow the rest of the test instructions and get sample WXR here: D105701-code

### CODE REVIEW

**Comment by jeroenpf** [comment]
I don't know if it is possible/easy to do, but instead of trying to find the attachment by filename, could we somehow store the WXR id & import id on attachments created through an import and check if an attachment with those id's already exists? If it exists, don't create it.

I see some issues with using the filename here:

- From what I can tell, the filename provided to the endpoint does not contain a directory map (e.g. `2023/03`) but the values in the `_wp_attached_file` do. Since you're doing a Regexp, files with similar names but different upload dates will be considered identical. I don't think it is completely uncommon for there to be multiple files with the same names but different upload dates. Furthermore, the filename may contain regexp modifiers. Not sure how the regexp compare feature works under the hood.
- Now, when a file exists, an error is returned. But we need to get a normal response with the uploaded filename and sizes (for backfilling purposes). So instead of returning an error, maybe return the existing resource and set a 409 status code.

ps. i just realize that using the import_id is not a great idea because it will be different for every import attempt. Using just the WXR post id should be fine. That should be unique on a site level.

**Comment by zaerl** [comment]
I can confirm that, as @jeroenpf suggested, a search inside `create_item()` for an attachment post with `unified_importer_id` equals to `$request['unified_importer_id']` is 100% proof that the attachment has already been uploaded in a previous import.

**Comment by vishnugopal** [comment]
Maybe if we have the source site URL: `{source-site-url}-{id}` as the key that would be best. But we're not sure if we get the source URL in the export file every time. If we don't then maybe `{hash of export file content}-{id}`.

**Comment by ouikhuan** [comment]
Closing this PR as this is not relevant anymore. Discussion here: p1679640263487699-slack-C01A60HCGUA

**Review by jeroenpf — APPROVED** [review]

File: projects/packages/import/src/endpoints/class-attachment.php [hunk 1]
```diff
@@ -96,6 +96,26 @@ public function register_routes() {
...
+
+			if ( \is_wp_error( $response ) ) {
+				return $response;
+			}
+
+			return new \WP_Error(
+				'attachment_exists',
+				__( 'The attachment already exists.', 'jetpack-import' ),
+				array(
+					'status'        => 409,
```
Comment (zaerl) [inline projects/packages/import/src/endpoints/class-attachment.php:112]: I checked some `WP_Error` that the WordPress core code generates in the `WP_REST_Controller` family of classes.

When a `*_exists` event occurs the `status` generated is usually a simple `400`. See, for example, `WP_REST_Menus_Controller::create_item`, `WP_REST_Terms_Controller::create_item`, and others.

I agree that technically speaking it is, in fact, a "conflict" but maybe we should follow the style found in dotorg code? What do you think about this?

**Review by zaerl — CHANGES_REQUESTED** [review]

**Review by zaerl — APPROVED** [review]
LGTM!

**Pull request merged**